zebra-state/src/service/queued_blocks.rs

```diff
@@ -286,6 +292,17 @@ impl SentHashes {
         self.update_metrics_for_block(block.height);
     }
 
+    /// Stores the finalized `block`'s hash, so it can be used to check if a
+    /// block is in the state queue.
+    ///
+    /// For more details see `add()`.
+    pub fn add_finalized_hash(&mut self, hash: block::Hash, height: block::Height) {
```

Comment:
I think this is implemented by `drain_queue_and_commit_finalized()`, because parent blocks are added before their child blocks. (And the checkpoint verifier only sends one block per height.)

For the non-finalized state, `send_ready_non_finalized_queued()` implements the same rule, but there can be multiple child blocks for the one parent block. (This forks the chain if the blocks are valid.)

Reply:
> I think this is implemented by drain_queue_and_commit_finalized(), because parent blocks are added before their child blocks.

Yes.

Updated the doc comments in [`de9730f`](https://github.com/ZcashFoundation/zebra/pull/6335/commits/de9730f08db982e1c952d1b3bf78e2c012657f9e).